src/analyze/analyze.c

```diff
@@ -188,14 +200,28 @@ static void free_unit_times(struct unit_times *t, unsigned n) {
         free(t);
 }
 
+static void subtract_timestamp(usec_t *a, usec_t b) {
+        if (*a > 0) {
+                assert(*a >= b);
+                *a -= b;
+        }
+}
+
+static int acquire_boot_times(sd_bus *bus, struct boot_times **bt);
```

Comment:
Static forward decls are ugly, IMO. As `acquire_boot_times()` doesn't call into `acquire_time_data()`, could you move the entire function up, probably even in an extra patch?